File: binary_search.java
import java.util.*;
public class binsh{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int size = 10;
        int[] nums= new int[size];

        for(int i=0;i<10;i++){
            nums[i]=in.nextInt();
        }
        int find = in.nextInt();
        System.out.println(search(nums,size,find));
    }
    public static int search(int arr[],int n,int key){
        int l=1;
        int h=n;
        while(l<=h){
            int mid = (l+h)/2;
            if(arr[mid]==key){
                return mid+1;
            }
            if(key<arr[mid]){
                h=mid-1;
            }
            else{
                l=mid+1;
            }
        }return 0;
    }
}

File: binary-search.cpp
#include<bits/stdc++.h>
using namespace std;

int binsrch(int flag,vector<int> foo , int r ,int l){
        if(r >=l ){
            int mid = l+(r-l) /2;
            if (foo[mid]==flag)
                return mid;
            if (foo[mid]>flag)
                return binsrch(flag,foo,mid -1 ,l);
            return binsrch(flag,foo,l,mid+1);
        }
        return -1;

};
int main(){
    vector<int> arr;
    int count = 5;
    int inp;
    for (int i = 0; i < count; i++)
    {
        cin>>inp;
        arr.push_back(inp);
    }
    int find;
    cin>>find;
    int ans = binsrch(find,arr,arr.size()-1,0);
    if(ans == -1)
        cout<<"Not Found";
    else
        cout<<ans;
}